containers/zammad/docker-entrypoint.sh

```diff
@@ -138,14 +140,19 @@ fi
 if [ "$1" = 'zammad-railsserver' ]; then
   test -f /opt/zammad/tmp/pids/server.pid && rm /opt/zammad/tmp/pids/server.pid
 
+  PUMA_OPTS=''
+  if (( ZAMMAD_WEB_CONCURRENCY > 1 )); then
+    PUMA_OPTS=" -w $ZAMMAD_WEB_CONCURRENCY"
+  fi
+
   check_zammad_ready
 
   cd "${ZAMMAD_DIR}"
 
-  echo "starting railsserver..."
+  echo "starting railsserver... with WEB_CONCURRENCY=${ZAMMAD_WEB_CONCURRENCY}"
 
   #shellcheck disable=SC2101
-  exec bundle exec rails server puma -b [::] -p "${ZAMMAD_RAILSSERVER_PORT}" -e "${RAILS_ENV}"
+  exec bundle exec rails server puma -b [::] -p "${ZAMMAD_RAILSSERVER_PORT} ${PUMA_OPTS}" -e "${RAILS_ENV}"
```

Comment:
using quotes per variable may prevent problematic rail server port containing whitespaces.

```suggestion
  exec bundle exec rails server puma -b [::] -p "${ZAMMAD_RAILSSERVER_PORT}" "${PUMA_OPTS}" -e "${RAILS_ENV}"
```

Reply:
Even without that the method `rails server puma` does not work:
```
bundle exec rails server puma -b [::] -p "3000" -w 2 -e production
Traceback (most recent call last):
        15: from bin/rails:9:in `<main>'
        14: from /usr/local/rvm/gems/ruby-2.6.6/gems/activesupport-5.2.4.6/lib/active_support/dependencies.rb:291:in `require'
        13: from /usr/local/rvm/gems/ruby-2.6.6/gems/activesupport-5.2.4.6/lib/active_support/dependencies.rb:257:in `load_dependency'
        12: from /usr/local/rvm/gems/ruby-2.6.6/gems/activesupport-5.2.4.6/lib/active_support/dependencies.rb:291:in `block in require'
        11: from /usr/local/rvm/gems/ruby-2.6.6/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
        10: from /usr/local/rvm/gems/ruby-2.6.6/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
         9: from /usr/local/rvm/gems/ruby-2.6.6/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
         8: from /usr/local/rvm/gems/ruby-2.6.6/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
         7: from /usr/local/rvm/gems/ruby-2.6.6/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
         6: from /usr/local/rvm/gems/ruby-2.6.6/gems/railties-5.2.4.6/lib/rails/commands.rb:18:in `<main>'
         5: from /usr/local/rvm/gems/ruby-2.6.6/gems/railties-5.2.4.6/lib/rails/command.rb:46:in `invoke'
         4: from /usr/local/rvm/gems/ruby-2.6.6/gems/railties-5.2.4.6/lib/rails/command/base.rb:69:in `perform'
         3: from /usr/local/rvm/gems/ruby-2.6.6/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
         2: from /usr/local/rvm/gems/ruby-2.6.6/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
         1: from /usr/local/rvm/gems/ruby-2.6.6/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
/usr/local/rvm/gems/ruby-2.6.6/gems/railties-5.2.4.6/lib/rails/commands/server/server_command.rb:139:in `perform': wrong number of arguments (given 2, expected 0) (ArgumentError)
        16: from bin/rails:9:in `<main>'
        15: from /usr/local/rvm/gems/ruby-2.6.6/gems/activesupport-5.2.4.6/lib/active_support/dependencies.rb:291:in `require'
        14: from /usr/local/rvm/gems/ruby-2.6.6/gems/activesupport-5.2.4.6/lib/active_support/dependencies.rb:257:in `load_dependency'
        13: from /usr/local/rvm/gems/ruby-2.6.6/gems/activesupport-5.2.4.6/lib/active_support/dependencies.rb:291:in `block in require'
        12: from /usr/local/rvm/gems/ruby-2.6.6/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
        11: from /usr/local/rvm/gems/ruby-2.6.6/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
        10: from /usr/local/rvm/gems/ruby-2.6.6/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
         9: from /usr/local/rvm/gems/ruby-2.6.6/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
         8: from /usr/local/rvm/gems/ruby-2.6.6/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
         7: from /usr/local/rvm/gems/ruby-2.6.6/gems/railties-5.2.4.6/lib/rails/commands.rb:18:in `<main>'
         6: from /usr/local/rvm/gems/ruby-2.6.6/gems/railties-5.2.4.6/lib/rails/command.rb:46:in `invoke'
         5: from /usr/local/rvm/gems/ruby-2.6.6/gems/railties-5.2.4.6/lib/rails/command/base.rb:69:in `perform'
         4: from /usr/local/rvm/gems/ruby-2.6.6/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
         3: from /usr/local/rvm/gems/ruby-2.6.6/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
         2: from /usr/local/rvm/gems/ruby-2.6.6/gems/thor-1.1.0/lib/thor/command.rb:20:in `run'
         1: from /usr/local/rvm/gems/ruby-2.6.6/gems/thor-1.1.0/lib/thor/command.rb:34:in `rescue in run'
/usr/local/rvm/gems/ruby-2.6.6/gems/thor-1.1.0/lib/thor/base.rb:525:in `handle_argument_error': ERROR: "rails server" was called with arguments ["-w", "2"] (Thor::InvocationError)
Usage: "rails server [puma, thin etc] [options]"
```

Tested this on a specific source code environment. 

However, if I adjust your command to my version it's working:
```
bundle exec puma -b tcp://[::]:3000 -w 2 -e production
[11294] Puma starting in cluster mode...
[11294] * Version 4.3.8 (ruby 2.6.6-p146), codename: Mysterious Traveller
[11294] * Min threads: 5, max threads: 30
[11294] * Environment: production
[11294] * Process workers: 2
[11294] * Preloading application
[11294] * Listening on tcp://[::]:11541
[11294] Use Ctrl-C to stop
[11294] - Worker 0 (pid: 11321) booted, phase: 0
[11294] - Worker 1 (pid: 11322) booted, phase: 0
```

That being said, I'm not sure if it's correct to listen to IPv6 stacks only.

Reply:
The variable change suggestion was just to get sure to not run in a problem with that.
Could also work but might prevent problems later.